sdk/dotnet/Pulumi/Deployment/Deployment_Invoke.cs

```diff
@@ -10,19 +10,18 @@ namespace Pulumi
 {
     public sealed partial class Deployment
     {
-        Task IDeployment.InvokeAsync(string token, ResourceArgs args, InvokeOptions? options)
+        Task IDeployment.InvokeAsync(string token, ResourceArgs? args, InvokeOptions? options)
```

Comment:
I've been changing tfgen not to pass nulls here, ready to open a PR for that. Avoiding nulls sounds like a better decision to me?

Reply:
https://github.com/pulumi/pulumi-terraform-bridge/pull/69

Reply:
i went back and forth on this (i made basically the exact same change to the bridge :)  But i ended updeciding i didn't like pushing this all the way out to every single caller. It seemed reasonable to me to say: you can pass me null and i'm ok with that here. 

I'm find with either approach.  one benefit of mine is that we dont have to update every resource provider.

Reply:
Agreed. I prefer avoiding defensive null checks, so I'd keep it non-nullable. Your workaround should stay for now until all providers are organically updated.